key_action_map.cpp

```diff
@@ -2,157 +2,195 @@
-        assert(action_type::none == m.to_action(sf::Keyboard::L));
+void test_key_action_map()//!OCLINT tests can be many
+{
+#ifndef NDEBUG // no tests in release
+  //It is possible to return an action type from a key being pressed
+  //The keys are for now the ones pressed by player1 (position 0 in the player index)
+  {
+    const key_action_map m; //Use player 1 as default for now
+    assert(m.to_action(sf::Keyboard::A) == action_type::turn_left);
+    assert(m.to_action(sf::Keyboard::D) == action_type::turn_right);
+    assert(m.to_action(sf::Keyboard::W) == action_type::accelerate);
+    assert(m.to_action(sf::Keyboard::S) == action_type::brake);
+    assert(m.to_action(sf::Keyboard::Q) == action_type::shoot);
+    assert(m.to_action(sf::Keyboard::Num1) == action_type::shoot_stun_rocket);
+  }
+  {
+    const key_action_map m = get_player_1_kam();
+    assert(m.to_action(sf::Keyboard::A) == action_type::turn_left);
+    assert(m.to_action(sf::Keyboard::D) == action_type::turn_right);
+    assert(m.to_action(sf::Keyboard::W) == action_type::accelerate);
+    assert(m.to_action(sf::Keyboard::S) == action_type::brake);
+    assert(m.to_action(sf::Keyboard::Q) == action_type::shoot);
+  }
+  {
+    const sf::Keyboard::Key key_to_go_left = sf::Keyboard::C;
+    const sf::Keyboard::Key key_to_go_right = sf::Keyboard::B;
+    const sf::Keyboard::Key key_to_accelerate = sf::Keyboard::A;
+    const sf::Keyboard::Key key_to_brake = sf::Keyboard::Z;
+    const sf::Keyboard::Key key_to_shoot = sf::Keyboard::E;
+    const key_action_map m(
+          key_to_go_left,
+          key_to_go_right,
+          key_to_accelerate,
+          key_to_brake,
+          key_to_shoot
+          );
+    assert(m.to_action(key_to_go_left) == action_type::turn_left);
+    assert(m.to_action(key_to_go_right) == action_type::turn_right);
+    assert(m.to_action(key_to_accelerate) == action_type::accelerate);
+    assert(m.to_action(key_to_brake) == action_type::brake);
+    assert(m.to_action(key_to_shoot) == action_type::shoot);
+  }
+  {
+    const key_action_map m = get_player_2_kam();
+    assert(m.to_action(sf::Keyboard::J) == action_type::turn_left);
+    assert(m.to_action(sf::Keyboard::L) == action_type::turn_right);
+    assert(m.to_action(sf::Keyboard::I) == action_type::accelerate);
+    assert(m.to_action(sf::Keyboard::K) == action_type::brake);
+    //assert(m.to_action(sf::Keyboard::Comma) == action_type::acc_backward);
+    assert(m.to_action(sf::Keyboard::U) == action_type::shoot);
+  }
+  {
+    const key_action_map m = get_player_1_kam();
+    assert( m.has_key(sf::Keyboard::A));
+    assert( m.has_key(sf::Keyboard::D));
+    assert( m.has_key(sf::Keyboard::W));
+    assert( m.has_key(sf::Keyboard::S));
+    assert( m.has_key(sf::Keyboard::Q));
+    assert(!m.has_key(sf::Keyboard::L));
+  }
 
-    }
-  #define FIX_ISSUE_282
-  #ifdef FIX_ISSUE_282
+  //Giving a key that is not in the map returns a action_type::none action
```

Comment:
I understand the design choice. I do think one day, that this will throw an exception, but maybe not. Let's see what the mediors think :-)

Reply:
Line 184 and the block that follows are not my work, I don't know what design choice has been made here.
I should stop touching the indentation/space of other people 😅